gregBorchers/Homework2GregBorchers-probs1-4NoCards.js

```diff
@@ -0,0 +1,359 @@
+  var numerator = 0;
+  var properFraction = "";
+
+  wholeNumber = Math.trunc( n / d );
+  numerator = n % d;
+
+  // general case format with non-zero values
+  properFraction = wholeNumber + " " + numerator + "/" + d;
+  
+  // handle special case formats for zero whole numbers
+  if (wholeNumber === 0) {
+    properFraction = numerator + "/" + d;
+  }
+  // handle special case formats for zero numerators
+  if (numerator === 0) {
+    properFraction = wholeNumber;
+  }
+
+  return properFraction;
+
+}
+
+// **************** Testing
+var n = 3;
+var d = 2;
+var properFractionString = fractionString2(n,d);
+console.log(properFractionString);
+
+n = 0;
+d = 6;
+var properFractionString = fractionString2(n,d);
+console.log(properFractionString);
+
+n = 4;
+d = 2;
+var properFractionString = fractionString2(n,d);
+console.log(properFractionString);
+
+//divide by zero... 
+n = 3;
+d = 0;
+var properFractionString = fractionString2(n,d);
+console.log(properFractionString);
+
+
+
+/*
+**2)** _(Moderate, 15%)_
+
+Rewrite each block below as the simplest equivalent you can discover.  Your
+equivalent should produce the same final conditions as the original code given
+the same initial conditions. In each case, if there is an undeclared variable,
+assume it has been declared earlier and set to an unknown value.
+
+//**2a)**
+*/
+
+var i;
+if ((n - Math.floor(n)) >= .5) {
+   i = Math.ceil(n);
+} else {
+  i = Math.floor(n);
+}
+
+// **************************  simplified form 2a)
+i = Math.trunc(n + .5);  // would be a simpler and shorter way to round n and store the rounded value in i
+
+
+//**2b)**
+
+var y, xIsFalse = (x? false : true);
+if (xIsFalse)
+   y = false;
+else
+   y = x;
+
+// **************************  simplified form 2b)
+var y = (x ? true : false);
+
+
+
+//**2c)**
+
+for (var count = 15 - i ; count < 15 ; count=count+1) {
+    i = i-1;  
+    console.log(i+1)
+}
+
+// **************************  simplified form 2c)
+// this code counts down from i to 1, printing counter each time.
+for (var count = i; count > 0; count--)
+{
+  console.log(count);
+}
+
+
+//**2d)**
+
+var x;
+if (a) {
+   if (b) {
+     x = 0;
+   } else {
+     x = 1;
+   }
+} else {
+  if (b) {
+    x = 1;
+  } else {
+    x = 2;
+  }
+}
+
+
+// **************************  simplified form 2b)
+// this should work for strings and valid numbers ( NaN breaks it)
+x = !a + !b;
+
+// ********************* problem 3)  Partridge Family Christmas song 
+/* 
+Write a function which returns a string containing the entire lyrics for the song "The
+Twelve Days of Christmas".  Make sure that your result is grammatically and
+typographically correct (include line breaks, commas, etc. where needed), but
+keep redundancy within your program to a minimum.  Don't just `console.log()`
+each line; return them together as a single string which includes a '\n' at each
+line break. You may want to use helper functions, loops, and/or arrays to store
+repeated elements. 
+*/
+
+
```

Comment:
3) You're right that it would be better to only include the "on the " x "day of Christmas...." once. There's a basic principle in Object-oriented programming called DRY: Don't Repeat Yourself. The idea is the more that your code is repetitive and redundant and says the same thing over and over (see what I did there?), the easier it is for mistakes to creep in and (more importantly) the harder it is to make changes later. So strive to keep your code dry as much as possible.